be/src/fs/fs_s3.cpp

```diff
@@ -307,19 +340,58 @@ static std::shared_ptr<Aws::S3::S3Client> new_s3client(const S3URI& uri, const F
             config.maxConnections = config::object_storage_max_connection;
         }
     } else {
-        if (!uri.endpoint().empty()) {
-            config.endpointOverride = uri.endpoint();
+        // resolve endpoint
+        auto it = opts._fs_options.find("fs.s3a.endpoint");
+        if (it != opts._fs_options.end() && !it->second.empty()) {
+            config.endpointOverride = it->second;
         } else if (!config::object_storage_endpoint.empty()) {
             config.endpointOverride = config::object_storage_endpoint;
+        }
+        // resolve region
+        it = opts._fs_options.find("fs.s3a.endpoint.region");
+        if (it != opts._fs_options.end() && !it->second.empty()) {
+            config.region = it->second;
+        } else if (!config::object_storage_region.empty()) {
+            config.region = config::object_storage_region;
+        }
+        // resolve ssl
+        it = opts._fs_options.find("fs.s3a.connection.ssl.enabled");
```

Comment:
does it mean, the s3a.endpoint should be not with `http://` or `https://`?

Reply:
no, just indicate that you need firstly retrieve values from the configuration item _fs_options, and then from config.

Reply:
what if the `fs.s3a.endpoint` configured with 'https://xxxx' and  `fs.s3a.connection.ssl.enabled` configured with `false`, or the vise versa, who will win?